src/vcpkg/configuration.cpp

```diff
@@ -535,6 +537,66 @@ namespace
 
 namespace vcpkg
 {
+    static Optional<std::string> get_baseline_from_git_repo(const VcpkgPaths& paths,
+                                                            StringView url,
+                                                            LocalizedString& error)
+    {
+        auto res = paths.git_fetch_from_remote_registry(url, "HEAD");
+        if (res.has_value())
+        {
+            return std::move(*res.get());
+        }
+        else
+        {
+            error = msg::format(msgUpdateBaselineGitError, msg::url = url)
+                        .appendnl()
+                        .append_raw(Strings::trim(res.error()));
+            return nullopt;
+        }
+    }
+
+    static constexpr StringLiteral BUILTIN_GIT_URL = "https://github.com/microsoft/vcpkg";
+
+    Optional<std::string> RegistryConfig::get_latest_baseline(const VcpkgPaths& paths, LocalizedString& error) const
```

Comment:
I'm not sure I understand the contract of this function. The others are one of two options:

1. The return value is engaged and the error is not set
2. The return value is disengaged and the error is set

which seem consistent. But this one has 

3. The return value is disengaged and the error is not set

and I don't know what to make of that.

(I guess what I'm looking for is a comment)